tool/src/main/java/org/wildfly/security/tool/FileSystemEncryptRealmCommand.java

```diff
@@ -0,0 +1,737 @@
+    private static final String SUMMARY_PARAM = "summary";
+    private static final String INPUT_REALM_LOCATION_PARAM = "input-location";
+    private static final String REALM_NAME_PARAM = "realm-name";
+    private static final String OUTPUT_REALM_LOCATION_PARAM = "output-location";
+    private static final String CREDENTIAL_STORE_LOCATION_PARAM = "credential-store";
+    private static final String SECRET_KEY_ALIAS_PARAM = "secret-key";
+    private static final String HASH_CHARSET_PARAM = "hash-charset";
+    private static final String HASH_ENCODING_PARAM = "hash-encoding";
+    private static final String ENCODED_PARAM = "encoded";
+    private static final String LEVELS_PARAM = "levels";
+    private static final String BULK_CONVERT_PARAM = "bulk-convert";
+    private static final String FILE_ARG = "file";
+    private static final String DIRECTORY_ARG = "directory";
+    private static final String NAME_ARG = "name";
+    private static final String DEFAULT_FILESYSTEM_REALM_NAME = "encrypted-filesystem-realm";
+    public static Supplier<Provider[]> ELYTRON_PASSWORD_PROVIDERS = () -> new Provider[]{
+            WildFlyElytronPasswordProvider.getInstance()
+    };
+
+    private final List<Descriptor> descriptors = new ArrayList<>();
+    private final List<String> PARAMS_LIST = new ArrayList<>(Arrays.asList(INPUT_REALM_LOCATION_PARAM, OUTPUT_REALM_LOCATION_PARAM));
+
+    private final Options options;
+    private final CommandLineParser parser = new DefaultParser();
+    private CommandLine cmdLine = null;
+    private boolean silentMode = false;
+    private boolean summaryMode = false;
+    private StringBuilder summaryString = null;
+    private boolean warningOccurred = false;
+
+    FileSystemEncryptRealmCommand() {
+        options = new Options();
+        Option option;
+
+        option = new Option("i", INPUT_REALM_LOCATION_PARAM, true, ElytronToolMessages.msg.cmdFileSystemEncryptInputLocationDesc());
+        option.setArgName(DIRECTORY_ARG);
+        options.addOption(option);
+
+        option = new Option("r", REALM_NAME_PARAM, true, ElytronToolMessages.msg.cmdFileSystemEncryptNewRealmDesc());
+        option.setArgName(DIRECTORY_ARG);
+        options.addOption(option);
+
+        option = new Option("o", OUTPUT_REALM_LOCATION_PARAM, true, ElytronToolMessages.msg.cmdFileSystemEncryptOutputLocationDesc());
+        option.setArgName(DIRECTORY_ARG);
+        options.addOption(option);
+
+        option = new Option("c", CREDENTIAL_STORE_LOCATION_PARAM, true, ElytronToolMessages.msg.cmdFileSystemEncryptCredentialStoreDesc());
+        option.setArgName(FILE_ARG);
+        options.addOption(option);
+
+        option = new Option("s", SECRET_KEY_ALIAS_PARAM, true, ElytronToolMessages.msg.cmdFileSystemEncryptSecretKeyDesc());
+        option.setArgName(FILE_ARG);
+        options.addOption(option);
+
+        option = new Option("h", HASH_CHARSET_PARAM, true, ElytronToolMessages.msg.cmdFileSystemEncryptHashCharsetDesc());
```

Comment:
I think all references to hash-charset can be removed right? I don't see the actual value actually being used anywhere.

Reply:
Removed